From dcbbf8b94831616203731ee8be9b40817aa6695f Mon Sep 17 00:00:00 2001
From: James Cowgill <james.cowgill@mips.com>
Date: Tue, 14 Nov 2017 12:18:37 +0000
Subject: [PATCH] MIPS: remove static specifier from register_pass_info struct

This fixes PR 82880 (where gcc --help --help hangs) by ensuring that if
mips_register_frame_header_opt is called twice, the second call to
register_pass gets a different instance of register_pass_info.

2017-11-14  James Cowgill <james.cowgill@mips.com>

	PR target/82880
	* config/mips/frame-header-opt.c (mips_register_frame_header_opt):
	Remove static specifier from f.
---
 gcc/config/mips/frame-header-opt.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/gcc/config/mips/frame-header-opt.c b/gcc/config/mips/frame-header-opt.c
index 76930792e92..0ebf377d046 100644
--- a/src/gcc/config/mips/frame-header-opt.c
+++ b/src/gcc/config/mips/frame-header-opt.c
@@ -99,7 +99,7 @@ void
 mips_register_frame_header_opt (void)
 {
   opt_pass *p = make_pass_ipa_frame_header_opt (g);
-  static struct register_pass_info f =
+  struct register_pass_info f =
     {p, "comdats", 1, PASS_POS_INSERT_AFTER };
   register_pass (&f);
 }
-- 
2.15.0

